### PR TITLE
Add docs for enabling flakes, recommend direnv extension

### PR DESCRIPTION
**Pull request by Eisfunke**
Some doc stuff for the new flake architecture.

While nix-env-selector *should* work through the flake-compat default.nix, we should rather recommend the direnv extension for VSCode so it uses the correct environment. That extension also tends to work more reliably and can be used in non-nix projects as well, so it's the better choice IMO.

### CODE REVIEW

File: Guide/installation.markdown [hunk 1]
```diff
@@ -112,6 +112,14 @@ Installing nix for IHP was done using [this guide](https://nathan.gs/2019/04/12/
 
 Note that nix can gradually grow to use several GB of disk space, especially after upgrading IHP. You can always run `nix-collect-garbage` in a `nix-shell` which will delete older/unused files.
 
+### Enabling flakes
```
Comment (mpscholten) [inline Guide/installation.markdown:115]: Maybe we can add, that this step is optional. If you disable flakes in your nix.conf, you can still run `ihp-new` and then still use the project (I've tried this yesterday myself)

Comment (mpscholten) [reply Guide/installation.markdown:115]: Yes when direnv is working, there's no manual need for `nix develop`

File: Guide/installation.markdown [hunk 2]
```diff
@@ -112,6 +112,14 @@ Installing nix for IHP was done using [this guide](https://nathan.gs/2019/04/12/
...
+
+IHP uses Nix flakes. While already widely used, they are not yet enabled by default by the Nix install scripts. To enable them, eithe edit `~/.config/nix/nix.conf` (to enable flakes just for your user) or `/etc/nix/nix.conf` (to enable flakes globally) and add the following line:
```
Comment (amitaibu) [inline Guide/installation.markdown:117]:
```suggestion
IHP uses Nix flakes. While already widely used, they are not yet enabled by default by the Nix install scripts. To enable them, either edit `~/.config/nix/nix.conf` (to enable flakes just for your user) or `/etc/nix/nix.conf` (to enable flakes globally) and add the following line:
```

Comment (Eisfunke) [reply Guide/installation.markdown]: Fixed, thanks!

File: Guide/installation.markdown (hunk 1 above)
Comment (Eisfunke) [reply Guide/installation.markdown:115]: I added some explaination there, what do you think?

**Review by mpscholten — APPROVED** [review]

**Pull request merged**